### PR TITLE
Re-structure code and add features before security testing

### PR DESCRIPTION
**Pull request by sullivan-james**
WARNING: before merging the apache configuration will need to be changed. This is because the wsgi.py script has moved. 

WARNING: before merging the following global system environment variable needs to be set (to replace filespace.txt and configure the filespace) - OCEAN_FILESPACE_PATH.

Please consult @sullivan-james before merging.

### CODE REVIEW

**Review by sullivan-james — COMMENTED** [review]
@olexandr-konovalov can you have a look at the changes in commit b18e38b please when you get a second. Is this what you meant by hashing for files? When a file gets uploaded a 32 bit hash code is created. I then have a method verify_hash() to re-calculate and compare.

**Comment by pgmccann** [comment]
Hi @sullivan-james, could you please make the changes necessary to address the issues highlighted at https://github.com/dolphin-acoustics-vip/database-management-system/security/dependabot ?

**Comment by sullivan-james** [comment]
Hi @pgmccann which issues are these? There are 3 issues still listed but these are features, not bugs. The link that you sent through does not lead anywhere. Cheers!

**Comment by sullivan-james** [comment]
@pgmccann I believe these dependency issues to have been fixed in 1518064

File: ocean/app/templates/encounter/encounter.html
Comment (pgmccann) [inline ocean/app/templates/encounter/encounter.html:1]: The console.log on line 71 should probably be removed, but doesn't prevent merging

**Review by pgmccann — APPROVED** [review]
I'm happy with this - the restructuring makes sense, and shouldn't present any significant challenges.

File: ocean/app/templates/encounter/encounter.html
Comment (sullivan-james) [reply ocean/app/templates/encounter/encounter.html:1]: Will address this in the next pull request

**Comment by sullivan-james** [comment]
Thanks! I have about 1000 unit tests with pytest that will be ready for a pull request tomorrow. This will accompany the stable code that I am happy for you to give to the security team @pgmccann

**Pull request merged**